rfcs/button-group-component.md

```diff
@@ -0,0 +1,43 @@
+# Button Group Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup>
+      <Button>Copy</Button>
+      <Button>Cut</Button>
+      <Button>Paste</Button>
+    </ButtonGroup>
+}
+```
+
+## Exports
+
+* `ButtonGroup`
+
+## `ButtonGroup` API
+
+* `children: (props: Props) => React.Node` - Required
+  This can be an SVG icon as well.
+* `ariaLabel: string` - Optional:
+  Accessible label
+* `disabled: boolean` - Optional:
+  Defines if the button group disabled.
+* `size: 'default' | 'compact'`
```

Comment:
Not sure why the other conversation got collapsed :(. But Do you think it's worth providing all the `Button` config API here and just say that it'll automatically be applied to all children `Button`? i.e. `KIND`, `SIZE`, `SHAPE`

Reply:
I think that makes sense - thoughts @nadiia @schnerd @blackswanny ?

Reply:
Yes, the same happens in other components. Props fall down to nested components.
For instance, Select sends props to each Option

Reply:
I like the idea of propagating the `kind`, `size`, `shape`, and related props down to buttons, though we might want to explicitly pass them rather than just passing `{...restProps}` down to each button.

Reply:
Totally makes sense since we already moving the `size` up